contracts/ANT-1/tests/evolve.test.ts

```diff
@@ -0,0 +1,47 @@
+import Arweave from 'arweave';
+import fs from 'fs';
+import path from 'path';
+import { JWKInterface, Warp } from 'warp-contracts';
+
+import { mineBlock } from '../../../tools/common/helpers';
+import { ANTState } from '../types';
+import { ANTDeployer } from '../utils';
+
+describe('Testing evolve...', () => {
+  const arweave: Arweave = global.arweave;
+  const wallets: Record<string, JWKInterface> = global.wallets;
+  const warp: Warp = global.warp;
+  const defaultOwner = Object.entries(wallets)[0];
+
+  let _srcTxId: string;
+
+  beforeEach(async () => {
+    const contractSource = fs.readFileSync(
+      path.join(__dirname, '../dist/contract.js'),
+      'utf8',
+    );
+    const srcTx = await warp.createSource(
+      { src: contractSource },
+      Object.values(wallets)[0],
+      true,
+    );
+    _srcTxId = await warp.saveSource(srcTx, true);
+    await mineBlock(arweave);
+  });
```

Comment:
does this need to be done `beforeEach`? it's a lot of overhead for each test to create a new source. you can likely just create one in `setup.jest.ts` and use the ID for testing interactions